common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -281,4 +284,44 @@ public Properties cryptoConf() {
   public long maxChunksBeingTransferred() {
     return conf.getLong("spark.shuffle.maxChunksBeingTransferred", Long.MAX_VALUE);
   }
+
+  /**
+   * Check if it is a shuffle client
+   * and avoid creating unnecessary event loops
+   * in the TransportChannelHandler
+   */
+  public boolean shuffleClient() {
```

Comment:
I think since this is so specialized and really only used from one place (the external shuffle service client) perhaps putting it here isn't that useful.  I think if we add a new constructor to the TransportContext with that optional parameter it might make more sense